### PR TITLE
Add styles panel to the new email editor [MAILPOET-5638]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

_N/A_

## Code review notes

- I  needed to refactor the settings sidebar slightly because it was not possible to use the `ComplementaryArea` component properly.
- I tried to prepare the data structure for the styles panel.
- The settings panel contains a few experimental components because I was inspired by the site editor.
- I am opened to discussion about each part of this PR.

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5638]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5638]: https://mailpoet.atlassian.net/browse/MAILPOET-5638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/EmailEditor/Engine/SettingsController.php [hunk 1]
```diff
@@ -94,26 +94,37 @@ public function getLayout(): array {
...
+        'width' => self::EMAIL_WIDTH,
+        'padding' => [
+          'bottom' => self::FLEX_GAP,
+          'left' => self::FLEX_GAP,
+          'right' => self::FLEX_GAP,
+          'top' => self::FLEX_GAP,
+        ],
+      ],
+      'colors' => [
+        'background' => '#000000',
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Engine/SettingsController.php]: What is the difference between layout.background and colors.background?

File: mailpoet/lib/EmailEditor/Engine/SettingsController.php [hunk 2]
```diff
@@ -113,10 +113,18 @@ public function getEmailStyles(): array {
         ],
       ],
       'colors' => [
-        'background' => '#000000',
+        'background' => '#ffffff',
       ],
       'typography' => [
       ],
+      // Value are only for purpose of displaying in the preview component in style sidebar
+      'elements' => [
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Engine/SettingsController.php:121]: I assume these are temporary and will be replaced by some set values or defaults. Am I correct?

File: mailpoet/assets/js/src/email-editor/engine/components/styles-sidebar/preview.tsx [hunk 3]
```diff
@@ -38,181 +41,191 @@ const secondFrame = {
...
+      slug: 'sample-secondary',
+      color: '#e5e2d3',
+    },
+    {
+      name: 'Sample Background',
+      slug: 'Sample-background',
+      color: '#a8c7a9',
+    },
+  ];
+  // https://github.com/WordPress/gutenberg/blob/7fa03fafeb421ab4c3604564211ce6007cc38e84/packages/edit-site/src/components/global-styles/hooks.js#L68-L73
```
Comment (costasovo) [inline mailpoet/assets/js/src/email-editor/engine/components/styles-sidebar/preview.tsx]: Where will we get these colors from?

I checked the link and if I understand correctly the primary is text color, the secondary is heading color and the background is document background. Am I right?

Shouldn't we set those in theme.json and apply them here as defaults? Later if a user sets their own, we should merge user-defined data with our theme.json data, and user's would win.

**Review by costasovo — COMMENTED** [review]
@lysyjan Nice job! I was not digging much into the components I assume we copied some stuff from the site editor. I was rather focused on the values and aded a few questions.

I also noticed that the back button for returning to the main panel differs from the site editor.

Back in our editor:

Screenshot 2024-03-01 at 10 41 56

Back in site editor:
Screenshot 2024-03-01 at 10 45 21

**Comment by lysyjan** [comment]
> I also noticed that the back button for returning to the main panel differs from the site editor.

@costasovo Those buttons are only for testing purposes. I'm expecting a replacement during the implementation of each section.

File: mailpoet/lib/EmailEditor/Engine/SettingsController.php (hunk 1 above)
Comment (lysyjan) [reply mailpoet/lib/EmailEditor/Engine/SettingsController.php]: The `layout.background` is used in the email editor wrapper, and the `colors.background` is used in the email.
example

I would prefer storing the layout background color in a different place, but I haven't found the right one yet.

File: mailpoet/lib/EmailEditor/Engine/SettingsController.php (hunk 2 above)
Comment (lysyjan) [reply mailpoet/lib/EmailEditor/Engine/SettingsController.php:121]: Yes, my expectation here is to merge the email editor default values and values set for an email post.

File: mailpoet/assets/js/src/email-editor/engine/components/styles-sidebar/preview.tsx (hunk 3 above)
Comment (lysyjan) [reply mailpoet/assets/js/src/email-editor/engine/components/styles-sidebar/preview.tsx]: @costasovo Based on our call, I added a commit using colors from the settings in the preview.
I hope I addressed all your questions during our call.

**Review by costasovo — APPROVED** [review]
@lysyjan Thanks for updating the values from the styles preview component

File: mailpoet/lib/EmailEditor/Engine/SettingsController.php (hunk 1 above)
Comment (costasovo) [reply mailpoet/lib/EmailEditor/Engine/SettingsController.php]: Ok. We will figure that out as we go.

**Pull request merged**